src/routing/mod.rs

```diff
@@ -312,17 +311,42 @@ impl Routing {
         dst: DstLocation,
         content: Bytes,
     ) -> Result<()> {
+        if let DstLocation::EndUser(EndUser::Client {
+            socket_id,
+            public_key,
+        }) = dst
+        {
+            let mut socket = None;
+            {
+                if let Some(socket_addr) = self.stage.state.lock().await.get_socket_addr(&socket_id)
+                {
+                    socket = Some(socket_addr.clone());
+                }
+            }
+
+            if let Some(socket_addr) = socket {
+                return self
+                    .send_message_to_client(socket_addr, ClientMessage::from(content)?)
+                    .await;
+            } else {
+                debug!(
+                    "Could not find socketaddr corresponding to socket_id {:?} and public_key {:?}",
+                    socket_id, public_key
+                );
+                debug!("Sending user message instead.. (Command::SendUserMessage)");
```

Comment:
Just to confirm, but does this make sense? Shouldn't we return an error instead?

Reply:
It makes sense, because if we are at section A, and want to send to a client which may be anywhere, we first check locally, and if we don't have the socket addr we relay it to the network.

Reply:
OK, but is that relaying implemented though? That is, when a node receives a `UserMessage` whose `dst` is `EndUser`, does it try to look up the socket addr and send the message to the client? Unless I'm missing something it doesn't seem to be implemented in this PR. Is that on purpose?

Reply:
The assumption in the above implementation is that the message is relayed to closest section by the next lines:

```
let command = Command::SendUserMessage { src, dst, content };
        self.stage.clone().handle_commands(command).await
``` 

Is it not?

Reply:
`approved.handle_user_message`, is where a relayed msg would turn up, if it iss deemed useful... And there it is raised to upper level by `Event::MessageReceived`.
So, there is no evaluation of if this is really a msg with enduser dst, and call to end_user_registry to send it using the socketaddr.

Where do you suggest that eval goes in? In `approved.handle_user_message`?